jobmaster/dm/message_agent.go

```diff
@@ -73,7 +80,14 @@ func (agent *MessageAgent) CreateWorker(ctx context.Context, taskID string, work
 	}
 	// TODO: deprecated subtask cfg
 	subTaskCfg := taskCfg.ToDMSubTaskCfg()
-	return agent.master.CreateWorker(workerType, subTaskCfg, 1)
+
+	var resourceConstraints []resourcemeta.ResourceID
```

Comment:
I think move this logic to WorkerManager's checkAndScheduleWorkers is more logical?